### PR TITLE
Refactor config with a dataclass

### PR DESCRIPTION
**Pull request by awav**
* Refactor config with the dataclass and enums
* Add an option for the positive transform bias
* Make the global config readable

### CODE REVIEW

File: gpflow/config/__config__.py [hunk 1]
```diff
@@ -1,106 +1,142 @@
...
-    return __config._float
+    return config().float
 
 
 def default_jitter():
-    return __config._jitter
+    return config().jitter
+
+
+def positive_minimum():
```
Comment (vdutor) [inline gpflow/config/__config__.py]: why not add `default` here to be consistent?

File: gpflow/config/__config__.py [hunk 2]
```diff
@@ -1,106 +1,142 @@
...
+    return config().positive_minimum
+
+
+def set_config(new_config: Config):
```
Comment (vdutor) [inline gpflow/config/__config__.py]: Would be nice to have a one line docstring here

File: gpflow/config/__config__.py [hunk 3]
```diff
@@ -1,106 +1,142 @@
...
+import enum
+import os
+from dataclasses import dataclass, field, replace
+from typing import Optional
 
 import numpy as np
 import tabulate
 import tensorflow as tf
 
 __all__ = [
```
Comment (vdutor) [inline gpflow/config/__config__.py]: A little example on how to use this config would be great! 
- how to get the default value
- how to change it

File: gpflow/config/__config__.py [hunk 4]
```diff
@@ -1,106 +1,142 @@
...
+    jitter: float = field(default_factory=lambda: default(_Values.JITTER))
+    positive_minimum: float = field(default_factory=lambda: default(_Values.POSITIVE_MINIMUM))
+    summary_fmt: str = _Values.SUMMARY_FMT.value
+
+
+def config() -> Config:
+    return __config
 
 
 def summary_fmt():
```
Comment (vdutor) [inline gpflow/config/__config__.py]: maybe add `default` here as well

File: gpflow/config/__config__.py (hunk 3 above)
Comment (awav) [reply gpflow/config/__config__.py]: Do you mean the documentation for the `config.py`?

**Pull request merged**